client/src/components/QA/QACardQuestions.jsx

```diff
@@ -6,53 +6,60 @@ import QAAddQuestionModal from './QAAddQuestionModal.jsx';
 
 const QACardQuestions = ({ questionsInfo, productName, currentProductId }) => {
 
-  // NOTE - keeping for refactor later on based on server changes for this path
-  // // answers array for current product
-  // const [answers, setAnswers] = useState([]);
-
   const [word, setWord] = useState('');
-  const [length, setLength] = useState(4);
+  const [temp, setTemp] = useState('');
+  const [length, setLength] = useState(2);
+  const [helpfulArray, setHelpfulArray] = useState([]);
 
-  // useEffect(() => {
-  // axios.get('/qa/questions/:question_id/answers')
-  //   .then(response => {
-  //     setAnswers(response.data);
-  //   })
-  //   .catch(err => err);
-  // }, []);
+  const APICallHelpful = (questionId) => {
+    if (helpfulArray.includes(questionId) === false) {
+      axios.put(`/qa/questions/:answer_id/helpful`, { question_id: questionId })
+        .then(info => console.log('info:', info))
+        .catch(err => err);
+      helpfulArray.push(questionId);
+    }
+  };
+
+  const APICallReport = (questionId) => {
+    axios.put(`/qa/questions/:question_id/report`, { question_id: questionId })
+      .then(info => console.log('info:', info))
+      .catch(err => err);
+  };
 
   const increaseCount = () => {
     event.preventDefault();
     setLength(length + 2);
   };
 
-  const filterQuestions = () => {
-    if (event.target.value.length < 3) {
+  const filterQuestions = (current) => {
+    setTemp(current);
+    if ((temp.length >= 0) && (temp.length < 3)) {
       setWord('');
-    } else {
-      setWord(event.target.value);
+    } else if (temp.length === 3) {
+      setWord(temp);
+      setLength(2);
```

Comment:
Why is the length being set to 2 here when it's being set to the questionInfo length later? I guess I don't get what `temp` and `length` do in the context of this func